### PR TITLE
use aws sdk to validate regions

### LINKED ISSUE #2619: Tie region validation to the SDK
We currently implement our own region validation mechanism: https://github.com/docker/distribution/blob/master/registry/storage/driver/s3-aws/s3.go#L106

This is not necessary anymore, and we should be able to use the SDK for that.

Either validating the region string proper: https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/

Or even using GetBucketRegion from the bucket name: https://docs.aws.amazon.com/sdk-for-go/api/service/s3/

The current situation makes our product regularly out of date and increases our maintenance burden.

### PR DESCRIPTION
**Pull request by davidswu**
closes #2619

Signed-off-by: David Wu <david.wu@docker.com>

### CODE REVIEW

**Comment by dmp42** [comment]
LGTM

**Comment by caervs** [comment]
LGTM

File: registry/storage/driver/s3-aws/s3.go [hunk 1]
```diff
@@ -105,25 +106,12 @@ type DriverParameters struct {
...
-		"ap-northeast-1",
-		"ap-northeast-2",
-		"sa-east-1",
-		"cn-north-1",
-		"us-gov-west-1",
-		"ca-central-1",
-	} {
-		validRegions[region] = struct{}{}
+	resolver := endpoints.DefaultResolver()
+	partitions := resolver.(endpoints.EnumPartitions).Partitions()
```
Comment (dmcgowan) [inline registry/storage/driver/s3-aws/s3.go]: I see there is also a `endpoints.DefaultPartitions()`, would this always return the same list?

Comment (davidswu) [reply registry/storage/driver/s3-aws/s3.go]: Yes let's use that instead

**Comment by dmp42** [comment]
Fixes #2619

**Pull request merged**